crates/cdk/src/wallet/wait.rs

```diff
@@ -0,0 +1,80 @@
+use cdk_common::wallet::{MeltQuote, MintQuote};
+use cdk_common::{Amount, Error, MeltQuoteState, MintQuoteState, NotificationPayload};
+use futures::future::BoxFuture;
+use tokio::time::{timeout, Duration};
+
+use super::{Wallet, WalletSubscription};
+
+#[allow(private_bounds)]
+enum WaitableEvent {
+    MeltQuote(String),
+    MintQuote(String),
+}
+
+impl From<&MeltQuote> for WaitableEvent {
+    fn from(event: &MeltQuote) -> Self {
+        WaitableEvent::MeltQuote(event.id.to_owned())
+    }
+}
+
+impl From<&MintQuote> for WaitableEvent {
+    fn from(event: &MintQuote) -> Self {
+        WaitableEvent::MintQuote(event.id.to_owned())
+    }
+}
+
+impl From<WaitableEvent> for WalletSubscription {
+    fn from(val: WaitableEvent) -> Self {
+        match val {
+            WaitableEvent::MeltQuote(quote_id) => {
+                WalletSubscription::Bolt11MeltQuoteState(vec![quote_id])
+            }
+            WaitableEvent::MintQuote(quote_id) => {
+                WalletSubscription::Bolt11MintQuoteState(vec![quote_id])
+            }
+        }
+    }
+}
+
```

Comment:
I think we should have another fn `mint_once_paid` to allow this flow: 

```
let quote = wallet.mint_quote(amount, None).await?;

let waiter = wallet.wait_for_payment(&quote.id, Duration::from_secs(10));
let proofs = waiter.await?;
```

With `wait_payment` it does not mint once it is paid so another fn call is needed.